### PR TITLE
[MERGE] mail: remove t-ref on component

### PR DESCRIPTION
**Pull request by alexkuhn**
Task-2694206

### CODE REVIEW

**Review by seb-odoo — CHANGES_REQUESTED** [review]
I don't like the approach in this PR, I think we should simply apply the same recipe I did in file uploader (which should be rebased on by the way):
use component to model into the child, access `child.component` through relation instead of `childRef.comp` in the parent

File: addons/mail/static/src/components/discuss_sidebar/discuss_sidebar.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
     <t t-name="mail.DiscussSidebar" owl="1">
         <div name="root" class="o_DiscussSidebar" t-attf-class="{{ className }}" t-ref="root">
             <div class="d-flex justify-content-center">
-                <button class="o_DiscussSidebar_startAMeetingButton btn btn-primary rounded" title="Start a meeting" t-on-click="discuss.onClickStartAMeetingButton" t-ref="startAMeetingButton">
+                <button class="o_DiscussSidebar_startAMeetingButton btn btn-primary rounded" title="Start a meeting" t-on-click="discuss.onClickStartAMeetingButton">
```
Comment (seb-odoo) [inline addons/mail/static/src/components/discuss_sidebar/discuss_sidebar.xml:7]: make a separate PR to remove this one if it's now useless

File: addons/mail/static/src/components/composer/composer.xml [hunk 2]
```diff
@@ -111,7 +112,7 @@
                                         <i class="fa fa-smile-o"/>
                                     </button>
                                     <t t-set="opened">
-                                        <EmojisPopover t-ref="emojisPopover" onEmojiSelection="_onEmojiSelection"/>
+                                        <EmojisPopover ref="'emojisPopover'" refs="refs" onEmojiSelection="_onEmojiSelection"/>
```
Comment (seb-odoo) [inline addons/mail/static/src/components/composer/composer.xml:115]: make a separate PR to introduce emojiPopoverView model if necessary (if I'm seeing correctly it's now the only component missing its view model in the diff of this PR)

File: addons/mail/static/src/components/discuss_sidebar/discuss_sidebar.xml (hunk 1 above)
Comment (alexkuhn) [reply addons/mail/static/src/components/discuss_sidebar/discuss_sidebar.xml:7]: PR with all useless t-refs removed: https://github.com/odoo/odoo/pull/81801

File: addons/mail/static/src/components/composer/composer.xml (hunk 2 above)
Comment (alexkuhn) [reply addons/mail/static/src/components/composer/composer.xml:115]: PR with use of popover view containing emoji list: https://github.com/odoo/odoo/pull/81877

**Comment by seb-odoo** [comment]
This is getting quite big, it's good to have separate commits, it would be nice if each commit had its own PR for ease of review and merge what is already ok.

**Comment by alexkuhn** [comment]
> This is getting quite big, it's good to have separate commits, it would be nice if each commit had its own PR for ease of review and merge what is already ok.

PR now split in 4 parts, all of them are listed on task.
One part is split in 2 commits just so each commit is easier to understand.

**Comment by alexkuhn** [comment]
> This PR is still referenced in the task, do you still plan to maintain it?

I thought there was still a final commit to keep at the end of merging all `t-ref` sub-tasks, but apparently there is none.
Thus I'm closing this PR, which is just useless.
